include/inviwo/core/util/interpolation.h

```diff
@@ -84,7 +85,15 @@ inline T linearVectorInterpolation(const T &a, const T &b, P x) {
 
 template <typename T, typename P>
 inline T Interpolation<T, P>::linear(const T &a, const T &b, P x) {
-    return glm::mix(a, b, x);
+    using VT = typename util::value_type<T>::type;
+    if constexpr (std::is_same_v<T, VT> || std::is_signed_v<VT> ) {
+        // GLM fails at correctly linearly interpolate between vectors of unsigned types when the
+        // difference is negative
+        return glm::mix(a, b, x);
+    } else {
+        using V = typename util::same_extent<T, double>::type;
+        return util::glm_convert<T>(glm::mix(util::glm_convert<V>(a), util::glm_convert<V>(b), x));
```

Comment:
why not just to a > b ? mix(a, b, x) : mix(b, a, 1-x)?

Reply:
will only work for scalars

Reply:
ahh but this is fixed in GLM, why are you adding a workaround here and not updating glm!?!

Reply:
https://godbolt.org/z/HtF2HY

Reply:
yes, it works with the latest glm, but way to cumbersome to update the submodule since it is a fork.